File: CompilationTest/TemplateString/results/TemplateString.swift
class TestClass {
  func testMethod() -> Void {
      let strVal = "str"
      let num = 1337
      let b = true
      let result = "before \(strVal), num: \(num), true: \(b) after"
      print(result)
      print("before \(strVal), num: \(num), true: \(b) after")
      
      let result2 = "before " + strVal + ", num: " + num + ", true: " + b + " after"
      print(result2)
  }
}

TestClass().testMethod()

File: CompilationTest/TemplateString/results/TemplateString.cpp
#include <iostream>
#include <string>

class TestClass {
  public:
    void testMethod() {
        auto str_val = std::string("str");
        int num = 1337;
        auto b = true;
        auto result = std::string("before ") + str_val + ", num: " + std::to_string(num) + ", true: " + (b ? "true" : "false") + " after";
        std::cout << result << std::endl;
        std::cout << std::string("before ") + str_val + ", num: " + std::to_string(num) + ", true: " + (b ? "true" : "false") + " after" << std::endl;
        
        auto result2 = std::string("before ") + str_val + std::string(", num: ") + std::to_string(num) + std::string(", true: ") + (b ? "true" : "false") + std::string(" after");
        std::cout << result2 << std::endl;
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}